core/store/src/db.rs

```diff
@@ -177,12 +177,27 @@ impl RocksDBOptions {
         self
     }
 
+    /// Opens the database either in read only or in read/write mode depending on the read_only
+    /// parameter specified in the store_config.
+    pub fn open(self, path: &Path, store_config: &StoreConfig) -> Result<RocksDB, DBError> {
```

Comment:
I meant this can stay `path: impl AsRef<Path>` but then do `let path = path.as_ref();` and the other two methods can just accept `&Path`.